client/src/main/java/org/asynchttpclient/channel/ChannelPoolPartitioning.java

```diff
@@ -23,12 +24,14 @@
         private final int proxyPort;
         private final boolean secured;
         private final String targetHostBaseUrl;
+        private final ProxyType proxyType;
 
-        public ProxyPartitionKey(String proxyHost, int proxyPort, boolean secured, String targetHostBaseUrl) {
+        public ProxyPartitionKey(String proxyHost, int proxyPort, boolean secured, String targetHostBaseUrl, ProxyType proxyType) {
```

Comment:
Is this really needed? I mean: would someone really have a SOCKS and an HTTP proxy on the same host?